domain/fhir/client.go

```diff
@@ -84,3 +87,20 @@ func (h httpClient) buildRequestURI(path string) string {
 	}
 	return h.url + path
 }
+
+func resolveResourcePath(resource interface{}) (string, error) {
+	data, err := json.Marshal(resource)
+	if err != nil {
+		return "", err
+	}
+	js := gjson.ParseBytes(data)
+	resourceType := js.Get("resourceType").String()
+	if resourceType == "" {
+		return "", fmt.Errorf("unable to determine resource type of %T", resource)
+	}
+	resourceID := js.Get("id").String()
+	if resourceType == "" {
+		return "", fmt.Errorf("unable to determine resource ID of %T", resource)
+	}
+	return resourceType + "/" + resourceID, nil
+}
```

Comment:
I guess this is fine but it feels a bit strange that we have to marshal/unmarshal JSON in order to get the path to the resource. Isn't there any way to create an interface which defines `ResourceType() string` and `ResourceID() string` for example?

Reply:
Good suggestion, unfortunately it's a (rather simplistic) library which just defines these structs and nothing more. So unless we add it ourselves, this is the easiest (laziest?) way for now.